### ISSUE TITLE
Not seamless on Firefox

### ISSUE THREAD

**Issue by jimbo1qaz** [regosen/Gapless-5#5]
Firefox supports perfect seamless looping, as shown here:
https://www.dropbox.com/s/buetaa1taw0ia2o/loop-firefox-example.7z?dl=1
(modified from https://bugzilla.mozilla.org/show_bug.cgi?id=654787 )

However, Gapless-5, contrary to its name, is not gapless on Firefox. The functionality not working is a bug.

Here are the logs I obtained by uncommenting some debug statements:

    loading track 0: sounds/sine.wav
    playing WebAudio

If I disable `playAudioFile` `->` `if (buffer != null)` with `&& false` , then the second if statement falls through (because audio == null).

**Comment by regosen** [comment]
Thanks for noticing this, and sorry for the slow response!

I looked at your link- unfortunately, while setting "source.loop = true;" allows the player to loop internally, it interferes with our ability to advance to the next track (used on sites like http://relisten.net/gapless to play gapless live albums).

That being said, I was able to replace the .onended() callback (which seems to fire too late in Firefox) with our own timer based on the remaining audio duration.  It seems to work well on Firefox without affecting the experience on Chrome or Safari.  I've just pushed the fix to master, let me know if it works for you!
